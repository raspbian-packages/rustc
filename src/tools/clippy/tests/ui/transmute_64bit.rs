//ignore-x86
//no-ignore-x86_64
#![feature(plugin)]
#![plugin(clippy)]

#[warn(wrong_transmute)]
fn main() {
    unsafe {
        let _: *const usize = std::mem::transmute(6.0f64);

        let _: *mut usize = std::mem::transmute(6.0f64);
    }
}
