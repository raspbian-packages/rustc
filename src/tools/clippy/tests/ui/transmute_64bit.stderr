error: transmute from a `f64` to a pointer
 --> $DIR/transmute_64bit.rs:9:31
  |
9 |         let _: *const usize = std::mem::transmute(6.0f64);
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `-D wrong-transmute` implied by `-D warnings`

error: transmute from a `f64` to a pointer
  --> $DIR/transmute_64bit.rs:11:29
   |
11 |         let _: *mut usize = std::mem::transmute(6.0f64);
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

